apps/apps.c

```diff
@@ -545,7 +545,7 @@ static int load_pkcs12(BIO *in, const char *desc,
     int len, ret = 0;
     PKCS12 *p12;
     p12 = d2i_PKCS12_bio(in, NULL);
-    if (p12 == NULL) {
+    if (p12 == NULL && desc != NULL) {
```

Comment:
A NULL pointer value in `desc ` should not shutdown error management, please ;) 
Better overwrite a NULL pointer to use a static default description like :
```
static const char desc_NA[] = "N/A";
...
if  (desc == NULL)
    desc = desc_NA
```

Reply:
@FdaSilvaYY, did you see my above remark?

> Moreover, when using this parameter, no checks had been done whether it is NULL, so I added such guards. This leads to the nice extra feature that error printing on file loading can be suppressed, which can be very handy for instance when trying to load files using different formats until (hopefully) this succeeds.

So consider this a feature, not a bug ;)
Still I like the suggested technique for adding a default description.

I plan to provide a further PR that makes use of this possibility, for instance as follows:

    static const char desc_default[] = "certificate";

    if  (desc == NULL)
        desc = desc_default;
    ERR_set_mark();
    cert = load_cert(file, format, pass, NULL);
    if (cert == NULL && (format == FORMAT_PEM || format == FORMAT_ASN1)) {
        format = (format == FORMAT_PEM) ? FORMAT_ASN1 : FORMAT_PEM;
        cert = load_cert(file, format, pass, NULL);
    }
    if (cert == NULL) {
        ERR_clear_last_mark();
        ERR_print_errors(bio_err);
        BIO_printf(bio_err, "error: unable to load %s from '%s'\n", desc, file);
    } else {
        ERR_pop_to_mark();
    }

This would provide a convenient second try in case the given or default PEM/DER format does not fit. I also have an even more useful variant that determines the format for the second try from the file name extension (and in case this yields the same format, flips PEM vs. DER as given above).

What you do you think about this?